src/test/compile-fail/maybe-bounds.rs

```diff
@@ -8,10 +8,9 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-trait Tr: ?Sized {} //~ ERROR `?Trait` is not permitted in supertraits
-                    //~^ NOTE traits are `?Sized` by default
+trait Tr: ?Sized {} //~ WARN default bound relaxed
 
-type A1 = Tr + ?Sized; //~ ERROR `?Trait` is not permitted in trait object types
-type A2 = for<'a> Tr + ?Sized; //~ ERROR `?Trait` is not permitted in trait object types
+type A1 = Tr + ?Sized; //~ WARN default bound relaxed
```

Comment:
hmm... I'm a little worried that this error has gone away. (I'd need to review the details about why its an error in the first place, but assuming that there was a good reason for it being an error, I assume that those same reason still hold here? Or was this just a case of "trait objects are implicitly unsized and thus it makes no sense to put `?Sized` in them" ?

Reply:
Trait objects previously had no default bounds, so the `dyn TraitA + ?TraitB` syntax wasn't useful. However now you can use it with `Move`, so `dyn Trait` implements `Move`, `dyn Trait + ?Move` does not.